benchmark/feldera-sql/run.py

```diff
@@ -367,32 +382,63 @@ def main():
         # Wait till the pipeline is completed
         start = time.time()
         last_processed = 0
+        last_metrics = 0
+        peak_memory = 0
         while True:
             stats = requests.get(f"{api_url}/v0/pipelines/{pipeline_name}/stats").json()
             elapsed = time.time() - start
-            processed = stats["global_metrics"]["total_processed_records"]
-            if processed > last_processed:
-                before, after = ('\r', '') if os.isatty(1) else ('', '\n')
-                sys.stdout.write(f"{before}Pipeline {pipeline_name} processed {processed} records in {elapsed:.1f} seconds{after}")
-                last_processed = processed
-            if stats["global_metrics"]["pipeline_complete"]:
-                break
+            if "global_metrics" in stats:
+                global_metrics = stats["global_metrics"]
+                processed = global_metrics["total_processed_records"]
+                peak_memory = max(peak_memory, global_metrics["rss_bytes"])
+                if processed > last_processed:
+                    before, after = ('\r', '') if os.isatty(1) else ('', '\n')
+                    sys.stdout.write(f"{before}Pipeline {pipeline_name} processed {processed} records in {elapsed:.1f} seconds{after}")
+                    if elapsed - last_metrics > metricsinterval:
+                        last_metrics = elapsed
+                        metrics_list = [pipeline_name, elapsed, global_metrics["rss_bytes"], global_metrics["buffered_input_records"], global_metrics["total_input_records"], global_metrics["total_processed_records"]]
+                        disk_index = 6
```

Comment:
this seems a little brittle maybe it should be `disk_index = len(metrics_list)+1` or similar?

Reply:
I'm going to go with this; the reason being that we don't necessarily know what order we are receiving things in, so we want to maintain the same order in the list of lists which will be output as a CSV.